### PR TITLE
Chef 14: Remove deprecated bootstrap options

### PR DESCRIPTION
**Pull request by tas50**
We said we would remove these in Chef 13 in the code. Lets do it for 14 instead.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: lib/chef/knife/bootstrap.rb [hunk 1]
```diff
@@ -307,10 +288,9 @@ def user_name
       end
 
       def bootstrap_template
-        # The order here is important. We want to check if we have the new Chef 12 option is set first.
-        # Knife cloud plugins unfortunately all set a default option for the :distro so it should be at
```
Comment (stevendanna) [inline lib/chef/knife/bootstrap.rb]: Does this imply any work for the cloud plugins or is this comment just out of date?

Comment (tas50) [inline lib/chef/knife/bootstrap.rb]: I'll have to check on that

**Comment by Ames0831** [comment]
I'm not sure what I'm doing?

**Comment by tas50** [comment]
This is going to require a bit of digging to figure out where else these are used

**Comment by lamont-granquist** [comment]
Mostly I'd suspect it would be knife-windows that would be mucking about with the internals enough of knife-bootstrap to break when these are removed...

**Comment by tas50** [comment]
The same options are deprecated in knife-windows so I don't see this impacting that. We'll nuke them from knife-windows and do a major version bump there as well.

File: lib/chef/knife/bootstrap.rb (hunk 1 above)
Comment (tas50) [reply lib/chef/knife/bootstrap.rb]: We do need to clean this all up in knife-cloud it appears.

**Comment by tas50** [comment]
I merged this on the CLI to fix the spec conflict. We've updated the knife-windows tests to not care about this CLI option. We should go through our other tools to make sure they're not trying to use these options anymore. They've been deprecated for years though and we said we were going to remove them in Chef 13.

**Pull request closed without merging**